packages/slice-machine/lib/mock/CustomType.ts

```diff
@@ -1,88 +1,54 @@
-import faker from "@faker-js/faker";
-import * as Widgets from "./misc/widgets";
-import { Tab } from "@lib/models/common/CustomType/tab";
-
-import { handleFields } from "./misc/handlers";
-
+import {
+  CustomType,
+  flattenWidgets,
+} from "@prismicio/types-internal/lib/customtypes/CustomType";
+import {
+  DocumentMockConfig,
+  DocWidgetMockConfig,
+  generateDocumentMock,
+  renderDocumentMock,
+} from "@prismicio/mocks";
 import { CustomTypeMockConfig } from "@lib/models/common/MockConfig";
-import { GroupSM } from "@slicemachine/core/build/models/Group";
-import { CustomTypeSM } from "@slicemachine/core/build/models/CustomType";
-
-interface Mock {
-  id: string;
-  uid: string | null;
-  type: string;
-  data: { [key: string]: unknown };
+import { buildWidgetMockConfig } from "./LegacyMockConfig";
+import { WidgetKey } from "@prismicio/types-internal/lib/documents/widgets";
+import {
+  CustomTypes,
+  CustomTypeSM,
+} from "@slicemachine/core/build/models/CustomType";
+
+function buildDocumentMockConfig(
+  model: CustomType,
+  legacyMockConfig: CustomTypeMockConfig
+): DocumentMockConfig {
+  const widgets = flattenWidgets(model);
```

Comment:
what does this do ?
Does it like list the widgets from both primary and sliceZone ?

Reply:
it flatten the widgets meaning that it returns an array of widgets without the `tabs` anymore. You can look in types-internal

Reply:
I see, so it's all about primary fields, it does not really represent the model in the end but I get that it's easier to read.

One question tho, if one 2 tabs you've got a widget with the same API ID, would that still work ?

Reply:
it's how custom types works, each api ID is unique in the whole custom type, tabs are here for wroom and then scrapped in the api so no problem here, always been the case.